### ISSUE TITLE
define actions within index_controls and let them be grouped in a actions_list

### ISSUE THREAD

**Issue by pjmuller** [avo-hq/avo#3135]
### Feature
In Avo 3.11 we don't need to define actions anymore in `def actions` and can directly use them in `def index_controls`

However the following code shows the actions as SEPARATE button and not grouped in the list
```ruby
class FoobarResource  < Avo::BaseResource
  def index_controls
     action Avo::Actions::BlaBla, arguments: {id: 1}
     action Avo::Actions::Haha, arguments: {id: 2}
     actions_list label: "lists"
  end
end
```
doesn't group them, they appear as top level entries

![Screenshot 2024-08-14 at 12 33 16](https://github.com/user-attachments/assets/473cc3ce-6930-4955-a0f8-b377845d6ffc)

### Current workarounds
the old flow does group them

```ruby
class FoobarResource  < Avo::BaseResource
  def actions
     action Avo::Actions::BlaBla, arguments: {id: 1}
     action Avo::Actions::Haha, arguments: {id: 2}
  end

  def index_controls
     actions_list label: "lists"
  end
end
```

⚠️ however since 3.11, the `arguments` gets LOST in this way (while they are remembered if defined within `index_controls` -> first code snippet of this issue)

**Comment by Paul-Bob** [comment]
> doesn't group them, they appear as top level entries

```ruby
class FoobarResource  < Avo::BaseResource
  def index_controls
     action Avo::Actions::BlaBla, arguments: {id: 1}
     action Avo::Actions::Haha, arguments: {id: 2}
     actions_list label: "lists" # list with all actions from `def actions`
  end

  def actions
    # empty
  end
end
```
`actions_list` picks the list from all the actions defined inside `def actions` so they appear as top-level entries as expected

> the old flow does group them
> ⚠️ however since 3.11, the `arguments` gets LOST in this way

We'll investigate it

**Comment by pjmuller** [comment]
@Paul-Bob 
1. is there a plan to make a different kind `actions_list` directive as to also create dropdowns "on the fly" within the `index_controls`
2. yes please, investigate why arguments are getting lost as it's essential for proper invoking of our actions

**Comment by Paul-Bob** [comment]
> is there a plan to make a different kind `actions_list` directive within `index_controls` to create dropdowns "on the fly" without having to define anything in `def actions`

Yes @pjmuller, check this issue https://github.com/avo-hq/avo/issues/2784

Ideally something like this:

```ruby
list do
  link_to "google", "https://google.com", ...options
  action ...options
end
``

**Comment by Paul-Bob** [comment]
Thanks for reporting this one @pjmuller. It's fixed on Avo `3.11.7`
